src/character.h

```diff
@@ -543,9 +543,29 @@ class Character : public Creature, public visitable<Character>
 
         units::mass weight_carried() const;
         units::volume volume_carried() const;
+
+        /// Sometimes we need to calculate hypothetical volume or weight.  This
+        /// struct offers two possible tweaks: a collection of items and
+        /// coutnts to remove, or an entire replacement inventory.
+        struct item_tweaks {
+            item_tweaks() = default;
+            item_tweaks( const std::map<const item *, int> &w ) :
+                without_items( &w )
+            {}
+            item_tweaks( const inventory &r ) :
+                replace_inv( &r )
+            {}
+            const std::map<const item *, int> *without_items = nullptr;
+            const inventory *replace_inv = nullptr;
```

Comment:
Any reasons that this are pointers? It could be references, or `optional<T>` if you want a "not used" value.

Reply:
They can't be references because they might not be present.  And optional values are expensive to copy.  I experimented with various ways of writing the code, and this seemed the cleanest.

Reply:
In that case, I recommend `optional<std::reference_wrapper<T>>`. It
- indicates that the value is optional,
- that (if it exists) is a reference (never null),
- is fast to copy as `reference_wrapper` is basically just a pointer,
- is easy to distinguish from a pointer (which may indicate ownership, or may require dynamic memory allocation, who knows).